### PR TITLE
[nrf temphack] Revert "[nrf fromlist] drivers: ...

### PR DESCRIPTION
**Pull request by e-rk**
... ieee802154: add option to retransmit a frame"

This reverts commit 24700b84247a0465ea2244268360130609a84ee4.

### CODE REVIEW

**Review by czeslawmakarski — APPROVED** [review]

**Review by edmont — APPROVED** [review]

**Comment by hubertmis** [comment]
What's the reason of reverting this commit?

**Comment by czeslawmakarski** [comment]
@hubertmis It was decided to remove retransmitting functionality due to the fact that it can leak security and introduce the limitation to the released solution.

File: drivers/ieee802154/ieee802154_nrf5.c [hunk 1]
```diff
@@ -567,20 +550,6 @@ static int nrf5_tx(const struct device *dev,
...
-	 * When frame encryption by the radio driver is enabled, the frame stored in
-	 * the tx_psdu buffer is:
-	 * 1) authenticated and encrypted in place which causes that after an unsuccessful
-	 *    TX attempt, this frame must be propagated back to the upper layer for retransmission.
-	 *    The upper layer must ensure that the exact same secured frame is used for
-	 *    retransmission
-	 * 2) frame counters are updated in place and for keeping the link frame counter up to date,
-	 *    this information must be propagated back to the upper layer
-	 */
-	memcpy(payload, nrf5_radio->tx_psdu + 1, payload_len);
```
Comment (edmont) [inline drivers/ieee802154/ieee802154_nrf5.c:581]: @e-rk I'm afraid we can't remove this `memcpy` because of the second reason.

**Review by edmont — CHANGES_REQUESTED** [review]

File: drivers/ieee802154/ieee802154_nrf5.c (hunk 1 above)
Comment (e-rk) [reply drivers/ieee802154/ieee802154_nrf5.c:581]: That's not good. This will still break the frames, because it might be encrypted multiple times.

Comment (e-rk) [reply drivers/ieee802154/ieee802154_nrf5.c]: I have removed this change for the time being.

Comment (edmont) [reply drivers/ieee802154/ieee802154_nrf5.c]: Thanks. We are assuming invalid retransmitted frames anyway, but at least operation without retransmissions won't be compromised.

**Review by edmont — APPROVED** [review]

**Comment by hubertmis** [comment]
> @hubertmis It was decided to remove retransmitting functionality due to the fact that it can leak security and introduce the limitation to the released solution.

I understand the reason, but I don't understand the solution. I think the patch should go to OpenThread to prevent retransmissions. Not to shim that simply relays requests from OpenThread to 802154 layer

**Comment by e-rk** [comment]
Closed in favor of an alternative solution.

**Pull request closed without merging**